### ISSUE TITLE
Heartbeat problem - Servers not listed every now and then

### ISSUE THREAD

**Issue by Fixmetal** [mvdevs/jk2mv#165]
Hello everybody,

I've got an issue with some jk2 servers I'm administering.
I've got 3 game servers on the same VM and sometimes they just are not listed on the server list. The firewall is opened on every UDP port involved for the game servers (just UDP), and at any time you can connect via `IP:PORT` anyway, but if you click the Refresh button on the server list typically one to two will not show their name, and then the behaviour would shift to the next server, like:

1. OK
2. NOK
3. NOK

< hit refresh button >

1. NOK
2. OK
3. NOK
and so on.

I've set these as sv_masters:
- masterjk2.ravensoft.com
- master.jk2mv.org
- master.jkhub.org
- masterjk3.ravensoft.com

Looking with `tcpdump` I see once in a while my VM would send ICMP (can't tell the type) over any heartbeat server:

```
# tcpdump -nni any icmp and host masterjk2.ravensoft.com or host master.jk2mv.org or host master.jkhub.org or host masterjk3.ravensoft.com
dropped privs to tcpdump
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on any, link-type LINUX_SLL (Linux cooked v1), capture size 262144 bytes
10:06:06.441803 IP <SERVER-IP>.52153 > 12.206.251.141.28060: UDP, length 27
10:06:06.441902 IP <SERVER-IP>.52153 > 78.47.50.174.28060: UDP, length 27
10:06:06.441987 IP <SERVER-IP>.52153 > 135.148.31.123.28060: UDP, length 27
10:06:06.442061 IP <SERVER-IP>.52153 > 12.206.251.141.28060: UDP, length 27
10:06:06.442125 IP <SERVER-IP>.52153 > 135.148.31.123.28060: UDP, length 27
10:06:06.531005 IP 135.148.31.123.28060 > <SERVER-IP>.52153: UDP, length 20
10:06:06.531079 IP 135.148.31.123.28060 > <SERVER-IP>.52153: UDP, length 20
10:06:06.531670 IP <SERVER-IP>.52153 > 135.148.31.123.28060: UDP, length 268
10:06:06.531749 IP <SERVER-IP>.52153 > 135.148.31.123.28060: UDP, length 268
10:06:06.552949 IP 12.206.251.141.28060 > <SERVER-IP>.52153: UDP, length 26
10:06:06.553702 IP <SERVER-IP>.52153 > 12.206.251.141.28060: UDP, length 812
10:06:52.073632 IP 12.206.251.141.28060 > <SERVER-IP>.43756: UDP, length 26
10:06:52.073779 IP <SERVER-IP> > 12.206.251.141: ICMP host <SERVER-IP> unreachable - admin prohibited filter, length 62
10:06:52.075234 IP 12.206.251.141.28060 > <SERVER-IP>.38348: UDP, length 26
10:06:52.075316 IP <SERVER-IP> > 12.206.251.141: ICMP host <SERVER-IP> unreachable - admin prohibited filter, length 62
10:06:52.097178 IP <SERVER-IP>.49317 > 12.206.251.141.28060: UDP, length 27
10:06:52.097333 IP <SERVER-IP>.49317 > 78.47.50.174.28060: UDP, length 27
10:06:52.097418 IP <SERVER-IP>.49317 > 135.148.31.123.28060: UDP, length 27
10:06:52.097493 IP <SERVER-IP>.49317 > 12.206.251.141.28060: UDP, length 27
10:06:52.185483 IP 135.148.31.123.28060 > <SERVER-IP>.49317: UDP, length 23
10:06:52.186074 IP <SERVER-IP>.49317 > 135.148.31.123.28060: UDP, length 274
10:06:52.208511 IP 12.206.251.141.28060 > <SERVER-IP>.49317: UDP, length 26
10:06:52.209672 IP <SERVER-IP>.49317 > 12.206.251.141.28060: UDP, length 868
10:06:58.078253 IP 12.206.251.141.28060 > <SERVER-IP>.45844: UDP, length 26
10:06:58.078403 IP <SERVER-IP> > 12.206.251.141: ICMP host <SERVER-IP> unreachable - admin prohibited filter, length 62
10:06:58.079932 IP 12.206.251.141.28060 > <SERVER-IP>.35075: UDP, length 26
10:06:58.080015 IP <SERVER-IP> > 12.206.251.141: ICMP host <SERVER-IP> unreachable - admin prohibited filter, length 62
10:06:58.594378 IP <SERVER-IP>.37156 > 12.206.251.141.28060: UDP, length 27
10:06:58.594485 IP <SERVER-IP>.37156 > 78.47.50.174.28060: UDP, length 27
10:06:58.594560 IP <SERVER-IP>.37156 > 135.148.31.123.28060: UDP, length 27
10:06:58.594635 IP <SERVER-IP>.37156 > 12.206.251.141.28060: UDP, length 27
10:06:58.594699 IP <SERVER-IP>.37156 > 135.148.31.123.28060: UDP, length 27
10:06:58.683160 IP 135.148.31.123.28060 > <SERVER-IP>.37156: UDP, length 22
10:06:58.683253 IP 135.148.31.123.28060 > <SERVER-IP>.37156: UDP, length 22
10:06:58.683876 IP <SERVER-IP>.37156 > 135.148.31.123.28060: UDP, length 269
10:06:58.683967 IP <SERVER-IP>.37156 > 135.148.31.123.28060: UDP, length 269
10:06:58.698103 IP 12.206.251.141.28060 > <SERVER-IP>.37156: UDP, length 26
10:06:58.698995 IP <SERVER-IP>.37156 > 12.206.251.141.28060: UDP, length 809
10:07:11.159592 IP <SERVER-IP>.50791 > 12.206.251.141.29060: UDP, length 27
10:07:11.159705 IP <SERVER-IP>.50791 > 135.148.31.123.29060: UDP, length 27
10:07:11.247726 IP 135.148.31.123.29060 > <SERVER-IP>.50791: UDP, length 22
10:07:11.248350 IP <SERVER-IP>.50791 > 135.148.31.123.29060: UDP, length 236
[...]
```

As you can see my VM is blocking outgoing ICMP traffic towards any heartbeat. This is the only thing I can trace and would make me think of a problem. No other logging issues anywhere else.
I'm looking for some clarification here about:
- Which heartbeat I should set
- Why is JK2MV supposed to send ICMP toward heartbeats - and which type should be (note to the less pro: ICMP is not just `ping`)
- Whether a game server is expected to open just for UDP protocol or both UDP+TCP protocols

Basically, what are the network requirements for a dedicated JK2MV server to run properly?

p.s. I won't disable the firewall on my VM, but even by disabling it, the issue is still present
p.p.s. I use SELinux, but audit.log isn't showing any denied messages, so that's not the issue

Thanks a lot,
Simone

**Comment by Fixmetal** [comment]
Hello @aufau ,
sorry for such a late response, but I've been terribly busy and eventually forgot about this issue.
At the time of writing I put  jk2mv dedicated server into a container and what I was seeking for was to make it run from there with podman+podman-compose, rootless. I've abandoned that after several digging around. The problem about podman is that it uses `slirp4netns` as a CNI network plugin which doens't allow a routable IP to be in place [0]. Given I was using Rocky Linux 8 and podman 4, I could have used `netavark` plugin, but that wasn't managed yet by podman-compose. So, basically I've abandoned the idea (I don't remember fully, but I've found some other difficulties by then).
FYI: I moved to an easiest rootless `systemd` approach in the end with some confinement thanks to that.

**TL;DR:** this was not jk2mv fault at all.

[0] https://github.com/containers/podman/blob/main/docs/tutorials/basic_networking.md#differences-between-rootful-and-rootless-container-networking
